src/codegen/doc/format.rs

```diff
@@ -54,121 +79,424 @@ fn get_language<'a>(entry: &'a str, out: &mut String) -> &'a str {
+) -> Option<String> {
+    if let Some((flag_info, member_info)) = env.analysis.flags.iter().find_map(|f| {
+        f.type_(&env.library)
+            .members
+            .iter()
+            .find(|m| m.c_identifier == symbol && !m.status.ignored())
+            .map(|m| (f, m))
+    }) {
+        Some(gen_member_doc_link(
+            flag_info.type_id,
+            &nameutil::bitfield_member_name(&member_info.name),
+            env,
+            in_type,
+        ))
+    } else if let Some((enum_info, member_info)) = env.analysis.enumerations.iter().find_map(|e| {
+        e.type_(&env.library)
+            .members
+            .iter()
+            .find(|m| m.c_identifier == symbol && !m.status.ignored())
+            .map(|m| (e, m))
+    }) {
+        Some(gen_member_doc_link(
+            enum_info.type_id,
+            &nameutil::enum_member_name(&member_info.name),
+            env,
+            in_type,
+        ))
+    } else if let Some(const_info) = env
+        .analysis
+        .constants
+        .iter()
+        .find(|c| c.glib_name == symbol)
+    {
+        Some(gen_const_doc_link(const_info))
+    } else {
+        None
+    }
+}
+
+// A list of types that are automatically ignored by the `find_type` function
+const IGNORED_C_TYPES: [&str; 6] = [
+    "gconstpointer",
+    "guint16",
+    "guint",
+    "gunicode",
+    "gchararray",
+    "GList",
+];
+/// either an object/interface, record, enum or a flag
+fn find_type(type_: &str, env: &Env) -> Option<String> {
+    if IGNORED_C_TYPES.contains(&type_) {
+        return None;
+    }
+
+    let type_id = if let Some(obj) = env.analysis.objects.values().find(|o| o.c_type == type_) {
+        Some(obj.type_id)
+    } else if let Some(record) = env
+        .analysis
+        .records
+        .values()
+        .find(|r| r.type_(&env.library).c_type == type_)
+    {
+        Some(record.type_id)
+    } else if let Some(enum_) = env
+        .analysis
+        .enumerations
+        .iter()
+        .find(|e| e.type_(&env.library).c_type == type_)
+    {
+        Some(enum_.type_id)
+    } else if let Some(flag) = env
+        .analysis
+        .flags
+        .iter()
+        .find(|f| f.type_(&env.library).c_type == type_)
+    {
+        Some(flag.type_id)
+    } else {
+        None
+    };
+
+    type_id.map(|ty| gen_symbol_doc_link(ty, env))
+}
+
+fn find_method_or_function_by_ctype(
+    c_type: Option<&str>,
+    name: &str,
+    env: &Env,
+    in_type: Option<(&TypeId, Option<LocationInObject>)>,
+) -> Option<String> {
+    find_method_or_function(
+        name,
+        env,
+        in_type,
+        |f| f.glib_name == name,
+        |o| c_type.map_or(true, |t| o.c_type == t),
+        |r| c_type.map_or(true, |t| r.type_(&env.library).c_type == t),
+        c_type.map_or(false, |t| t.ends_with("Class")),
+    )
+}
+
+/// Find a function in all the possible items, if not found return the original name surrounded with backticks
```

Comment:
You have the right to add dots at the end of your sentences. It's even more important here since it's one paragraph since there is no empty line after. ;)